### ISSUE TITLE
[Bug] Can't resize an image AFTER it has been inserted into a note. 

### ISSUE THREAD

**Issue by Drakemoor** [xRyul/obsidian-image-converter#82]
As the title say, I can not resize an image AFTER it has been inserted into a note, using the right click menu. 

1. Insert an image into a note. - it converts and is resized as per the plugin settings
2. Right click on the image and select: 
![image](https://github.com/user-attachments/assets/5e658777-45f5-4578-9733-c30b6a391b82)
'Resize Image' from the menu.
3. Add some value into the modal: 
![image](https://github.com/user-attachments/assets/afee5758-e575-4c6f-a072-372184b0a5d3)
4. Click 'Submit'
5. Nothing happens. 

Please let me know if I can provide more info. 

Thank you.

**Comment by xRyul** [comment]
Can you verify if the conversion was successful? e.g.: right-click on the image link and see if you can find it in your vault. We need to make sure that image had time to render, and we are not dealing with cached version of it. 

![image](https://github.com/user-attachments/assets/c251d281-e49c-45df-acaf-63a8e6e54301)


https://github.com/user-attachments/assets/0ef62061-2bb2-4801-8b98-b43c238ff1b2

**Comment by Drakemoor** [comment]
Yes, the conversion is good. I have checked that before I posted the bug report. I also monitor the actual file on the hdd. It's only when I try to resize it after that it doesn't work. 

BTW, the videos you are posting here and on the home github page of the plugin doesn't show up for Mr.

**Comment by xRyul** [comment]
hmm does the same happen on sanboxed/clean vault?

**Comment by Drakemoor** [comment]
Yes, it's acting the same in the sandbox vault. 

Attached to this are zip files with the settings of 'Image-Converter' and 'Custom Attachments Location' plugins, that I'm using, plus an image that I use. 

[image-converter.zip](https://github.com/user-attachments/files/17655726/image-converter.zip)
[obsidian-custom-attachment-location.zip](https://github.com/user-attachments/files/17655727/obsidian-custom-attachment-location.zip)
[Clipboard Image.zip](https://github.com/user-attachments/files/17655730/Clipboard.Image.zip)

Let me know if I can provide any more info or help in any other way. 

Thank you.

**Comment by xRyul** [comment]
> Yes, the conversion is good. I have checked that before I posted the bug report. I also monitor the actual file on the hdd. It's only when I try to resize it after, that it doesn't work.

So image is resized correctly? I am not sure what is the issue then? It resizes fine on my end too. Both during the drop, paste and via right click menu. It also looks fine when I look it up in the folder, it shows the new dimensions. 

By any chance do you have "|size" applied to the link?  And what size are you trying to resize it to via right click? If resizing from e.g. 800px to 700px or even 600px, difference will be minimal. In particular when you combine it with | size. Because | size will freeze the image.

**Comment by Drakemoor** [comment]
The image is resizing fine during drop and paste.  

The problem is I cannot resize it after via right click menu.  

Yes, I have "|size" applied to the link. Here is an example of image I'm trying to resize via right click menu. 
```
![[2024-11-09-20241104_125956.webp|250]]
```
I'm trying to resize from 4000x2252 to 800px. And I watch the image dimensions of the file on the HDD, not only in the note itself. 

The right click resize command do not work for me. 

Thank you for looking into this.

**Comment by Drakemoor** [comment]
hi,

Any news about this? 

If I can provide more info just let me know. 

Thanks

**Comment by xRyul** [comment]
I am unable to reproduce it. And none else reported this, which makes it hard to pinpoint what might be causing it. 

Latest release changed how images get rendered. Thus the last thing you can try.

1. Dont use mobile. Open it on Desktop machine.
2.. Disable all plugins and custom snippets. 
3.. Drag and drop image from Desktop into the note.
4.. If "| size" got applied -> remove it. 
5. Now try resize it.
6. If changes appear? great. If not? Maybe someone can create a PR for this.

**Comment by Drakemoor** [comment]
Hi,

Thank you for trying to help. 

I followed the steps you listed. I don't use mobile, I never had. 
1. I open a new Sandbox Vault
2. I install "image converter" plugin
3. there are no other plugins installed
4. I drag and drop the image from my desktop into a note.
5. There is no "|size" applied
6. I do the following:  
	1. right-click the image
	2. click "Resize Image"
	3. specify the 'width': 100
	4. click submit
	5. Nothing happens

The image size stays the same.  
The actual file on the HDD does NOT change.  
There is nothing in Obsidian console.  

Attached is the image I'm using.  
[Clipboard Image (1)_2024_12_01_14-08-00.zip](https://github.com/user-attachments/files/17970038/Clipboard.Image.1._2024_12_01_14-08-00.zip)

Please let me know if there is something else I can try.  

What is "PR"?

**Comment by xRyul** [comment]
Hey, Could you please try a new command:

<img width="218" alt="image" src="https://github.com/user-attachments/assets/14963768-b36e-4a9b-9b5e-8a8be15c5294" />


And then use following settings:

<img width="413" alt="image" src="https://github.com/user-attachments/assets/d45c97e3-c78d-414e-974c-151220d39583" />

Then click "Process" and manually refresh the note e.g.: click back and forth icon at the top, or just click on a another note and then back to your original one.

<img width="58" alt="image" src="https://github.com/user-attachments/assets/e6e9df36-faaa-4616-978a-4b4d654b563f" />

**Comment by IainBapty** [comment]
I'm not the original reporter of this bug but found this thread today when encountering the same issue. Resizing appeared to not be making any changes to the file on disk. I've just updated to the latest and tried the approach referenced by @xRyul which **has worked fine for me!** Thank you.

Off topic but is the rename option in your right-click menu a plugin?

**Comment by Drakemoor** [comment]
Thanks @xRyul , the 'Convert/compress' option works as intended. ❤  

The original 'Resize" command still does nothing for me. Are you considering removing it from the right click menu? 

Thanks again for the great plugin.  

BTW the new additions to 'Rename' and 'Move' images are greatly appreciated. 🙇‍♂️

**Comment by Drakemoor** [comment]
One additional observation.  

If the Advanced option `Allow larger file sizes after conversion` is OFF, and I try to enlarge an image, the process returns no feedback to the user. It would be helpful if it says something along "The image have not been processed because the `Allow larger file sizes after conversion` setting is turned off".

**Comment by xRyul** [comment]
@IainBapty rename via right-click are part of Image Converter, but only work with `Native Menus` disabled `Obsidian Settings > Appearance > Advanced > Native menus`

@Drakemoor It will be still there, but I am planning on making it more visual and allow it to resize canvas of the image too. 

`Allow larger file sizes after conversion` when turned OFF it uses older algorithm, so if on drop/paste it couldn't achieve any significant reduction in file size it won't show anything. How much reduction it will or will not do, will depend on file format and image content itself.
